cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
Ok I noticed that you modified the kube dns addon for vsphere here. I did not look if you modified more of the vsphere deployment so I cc'ed @imkin if he wants to look over this PR